### PR TITLE
Add unique social title and description 

### PR DESCRIPTION
**Pull request by ykmsd**
# PR Description
Currently the open graph title for every docs page is "StackBlitz Docs" and the description is "Discover how...". This PR aims to fix it. 

If I understand [this issue](https://github.com/vuejs/vitepress/issues/975) correctly, we have to set the og:title for each pages to have a unique title (feels repetitive but couldn't find another way). For the description I mostly used the first paragraph of each pages and made a few for the pages didn't have an introduction. It would be great if @sylwiavargas  could check these descriptions make sense 🙏 (especially [the ones I wrote](https://github.com/stackblitz/docs/pull/81/commits/72f77d9379e219c2146c6c951569ae011c8d0562) 🥲)

This PR fixes the issue number:  [SBPF-2866](https://linear.app/stackblitz/issue/SBPF-2866/docs-social-cards-have-to-have-descriptive-title-description)

### Summary of my changes and explanation of my decisions

- Added og:title info in each posts to set a unique title
- Added a description since most of the pages didn't have a description and this is why the default one was used. 

### Self-check

Please check all that apply:

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my code or content update and edited it to the best of my abilities
- [x] I have commented my code, particularly in hard-to-understand areas; my comments are concise

### Contributors list

Would you like your contribution to be celebrated? Please check all that apply:

- [ ] I would like to be featured on the future contributors list in the README. If so, please tell us:
  - what name you'd like to be shown there?
  - we usually link github profile pic -- is that ok? If not, provide another url: 
  - we usually link your github profile but if you have a portfolio page, provide a link:

- [ ] I would like to get a shoutout in the next monthly updates post. If so, please provide your twitter handle (or we will link to your GitHub profile).

⚡️ ⚡️ ⚡️  Thank you for contributing to StackBlitz ⚡️ ⚡️ ⚡️

### CODE REVIEW

File: docs/codeflow/codeflow-faq.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 title: Codeflow FAQ
+description: 'This page addresses the frequently asked questions concernin: current features, security, planned features, pricing and access.'
```
Comment (ykmsd) [inline docs/codeflow/codeflow-faq.md:3]: 📝 Put it insite the quote to escape a colon `:`

File: docs/guides/user-guide/importing-projects.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ---
 title: Importing projects
+description: This page outlines how you can import projects to StackBlitz.
```
Comment (ykmsd) [inline docs/guides/user-guide/importing-projects.md:3]: ❓ This page doesn't have an intro so I needed to make it.

File: docs/platform/api/javascript-sdk-options.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 ---
 title: SDK Options Reference
+description: This page provides the available options for StackBlitz SDK projects.
```
Comment (ykmsd) [inline docs/platform/api/javascript-sdk-options.md:3]: ❓ This page doesn't have an intro so I needed to make it.

**Comment by ykmsd** [comment]
When I check locally I only see one `og:title` but when I check [the preview link ](https://deploy-preview-81--stackblitz-docs.netlify.app/) I see `og:title` twice (the default one + individual one). This is the same for `og:image` though, is it expected? 🤔

File: docs/guides/user-guide/importing-projects.md (hunk 2 above)
Comment (sylwiavargas) [reply docs/guides/user-guide/importing-projects.md:3]: Thank you!!

File: docs/platform/api/javascript-sdk-options.md (hunk 3 above)
Comment (sylwiavargas) [reply docs/platform/api/javascript-sdk-options.md:3]:
```suggestion
description: Modify your StackBlitz project with SDK options.
```

**Review by sylwiavargas — APPROVED** [review]
Thank you, Yuka, for all your effort and attention to detail. Left just one comment but otherwise, it's good to go provided we address the `og:title` from your previous question :)

**Comment by fvsch** [comment]
> When I check locally I only see one `og:title` but when I check [the preview link ](https://deploy-preview-81--stackblitz-docs.netlify.app/) I see `og:title` twice (the default one + individual one). This is the same for `og:image` though, is it expected? 🤔 @fvsch can maybe confirm 🙏

So it looks like:

1. The static HTML output for the page is correct, there's only one `og:image` or `og:title` and it's the one provided in the page's Markdown front-matter.
2. The JS code that hydrates the page seems to update the page's `<head>` and we end up with duplicate `og:image` and `og:title` values.

I'd say the second part is a bug in Vitepress that we can report but probably not fix on our side. It's perhaps not critical either, as long as the HTML response has the correct data, since most bots will not run a full browser and will only look at the HTML.

**Comment by fvsch** [comment]
One fix we could do on our side is removing the default `og:title` and `og:image` from the theme configuration, and moving those values to the home page's Markdown front-matter.

**Comment by ykmsd** [comment]
@fvsch Hmm so am I correct to move `og:title` to `index.md`? Seems like Devon already moved `og:image` there [in the last PR](https://github.com/stackblitz/docs/pull/76/commits/6964f6769b60cc2a296d111cf07efbb9a81526a1) but I still see two og:title and og:image and I'm not sure how it works, the head of index.md will go to every pages? 🤔 When I post a preview link on Slack, the image shows the correct one but the title is the default one 🤨
e.g. https://deploy-preview-81--stackblitz-docs.netlify.app/codeflow/what-is-codeflow
Screenshot 2023-01-11 at 13 55 21

**Comment by fvsch** [comment]
> the head of index.md will go to every pages?

No, it will only affect the home page.

Pages that don't explicitly define an `og:title` value in their Markdown front-matter will _not_ have an `og:title`. Which I'm okay with, because generic `og:title` values are not super useful; social networks will fall back to the title element anyway.

**Comment by ykmsd** [comment]
Seems like I needed to do hard refresh and they now have only one `og:title` so I guess it's good to go, I will come back to you if it doesn't work in production ^^ Merci @fvsch!

**Pull request merged**